src/QueryReflection/QueryReflection.php

```diff
@@ -187,13 +188,21 @@ private function resolveQueryExpr(Expr $queryExpr, Scope $scope): ?string
      */
     private function resolveQueryStringExpr(Expr $queryExpr, Scope $scope): ?string
     {
-        if ($queryExpr instanceof Expr\MethodCall && $queryExpr->name instanceof Identifier) {
-            $classReflection = $scope->getClassReflection();
-
-            // XXX atm we only support inference-placeholder for method calls within the same class
-            if (null !== $classReflection && $classReflection->hasMethod($queryExpr->name->name)) {
-                $methodReflection = $classReflection->getMethod($queryExpr->name->name, $scope);
+        if ($queryExpr instanceof Expr\CallLike) {
+            $methodReflection = null;
+            if ($queryExpr instanceof Expr\StaticCall) {
+                if ($queryExpr->class instanceof Name && $queryExpr->name instanceof Identifier) {
+                    $classType = $scope->resolveTypeByName($queryExpr->class);
+                    $methodReflection = $scope->getMethodReflection($classType, $queryExpr->name->name);
+                }
+            } elseif ($queryExpr instanceof Expr\MethodCall && $queryExpr->name instanceof Identifier) {
+                $classReflection = $scope->getClassReflection();
+                if (null !== $classReflection && $classReflection->hasMethod($queryExpr->name->name)) {
+                    $methodReflection = $classReflection->getMethod($queryExpr->name->name, $scope);
+                }
+            }
```

Comment:
we still don't support all calls possible by php, but should be enough for my use-case in REDAXO and `rex_sql::getTablePrefix()` for now